sdk/core/azure-core/src/main/java/com/azure/core/models/CloudEvent.java

```diff
@@ -0,0 +1,358 @@
+
+    /*
+     * Event data specific to the event type, encoded as a base64 string.
+     */
+    @JsonProperty(value = "data_base64")
+    private byte[] dataBase64;
+
+    /*
+     * Type of event related to the originating occurrence.
+     */
+    @JsonProperty(value = "type", required = true)
+    private String type;
+
+    /*
+     * The time (in UTC) the event was generated, in RFC3339 format.
+     */
+    @JsonProperty(value = "time")
+    private OffsetDateTime time;
+
+    /*
+     * The version of the CloudEvents specification which the event uses.
+     */
+    @JsonProperty(value = "specversion", required = true)
+    private String specVersion;
+
+    /*
+     * Identifies the schema that data adheres to.
+     */
+    @JsonProperty(value = "dataschema")
+    private String dataSchema;
+
+    /*
+     * Content type of data value.
+     */
+    @JsonProperty(value = "datacontenttype")
+    private String dataContentType;
+
+    /*
+     * This describes the subject of the event in the context of the event
+     * producer (identified by source).
+     */
+    @JsonProperty(value = "subject")
+    private String subject;
+
+    @JsonIgnore
+    private Map<String, Object> extensionAttributes;
+
+    /*
+     * Cache serialized data for getData()
+     */
+    @JsonIgnore
+    private BinaryData binaryData;
+
+    private CloudEvent() {
+
+    }
+
+    public CloudEvent(String source, String type, Object data) {
+        this(source, type, data, null);
+    }
+
+    public CloudEvent(String source, String type, Object data, String dataContentType) {
+        this.source = source;
+        this.type = type;
+        if (data instanceof byte[]) {
+            this.dataBase64 = Base64.getEncoder().encode((byte[]) data);
+        } else {
+            this.data = data;
+        }
+        this.dataContentType = dataContentType;
+        this.id = UUID.randomUUID().toString();
+        this.specVersion = CloudEvent.SPEC_VERSION;
+    }
+
+    /**
+     * Deserialize the {@link CloudEvent} from a JSON string.
+     * @param cloudEventsJson the JSON payload containing one or more events.
+     *
+     * @return all of the events in the payload deserialized as {@link CloudEvent CloudEvents}.
+     * @throws IllegalArgumentException if the input parameter isn't a JSON string for a cloud event or an array of it.
+     */
+    public static List<CloudEvent> fromString(String cloudEventsJson) {
+        try {
+            List<CloudEvent> events = Arrays.asList(DESERIALIZER.deserialize(
+                new ByteArrayInputStream(cloudEventsJson.getBytes(StandardCharsets.UTF_8)),
+                TypeReference.createInstance(CloudEvent[].class)));
+            for (CloudEvent event : events) {
+                if (event.getSource() == null || event.getType() == null) {
+                    throw LOGGER.logExceptionAsError(new IllegalArgumentException(
+                        "'source' and 'type' are mandatory attributes for a CloudEvent. " +
```

Comment:
EG service doesn't require source (not sure about type), so customers may not be able to use this type with data that they are publishing through the REST endpoint. I'm not saying that we definitely shouldn't have the validation, but we probably need to make a call as to what the right thing to do is here across languages.

Reply:
"source" is required in the CloudEvent spec. We need to sync among all languages.
https://github.com/cloudevents/spec/blob/v1.0.1/spec.md#required-attributes

Reply:
Now users have two options.
`fromString(String cloudEventsJson)` and `fromString(String cloudEventsJson, bool skipValidation)`